front/src/modules/rollingStock/components/RollingStockEditor/CurveSpreadsheet.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { Dispatch, SetStateAction, useMemo } from 'react';
+import Spreadsheet, { createEmptyMatrix } from 'react-spreadsheet';
+import { emptyStringRegex, onlyDigit } from 'utils/strings';
+import { replaceElementAtIndex } from 'utils/array';
+import { useTranslation } from 'react-i18next';
+
+import type { CellBase, Matrix } from 'react-spreadsheet';
+import type {
+  ConditionalEffortCurveForm,
+  EffortCurveForm,
+  EffortCurveForms,
+} from 'modules/rollingStock/types';
+
+type CurveSpreadsheetProps = {
+  selectedCurve: ConditionalEffortCurveForm;
+  selectedCurveIndex: number;
+  effortCurves: EffortCurveForms | null;
+  setEffortCurves: Dispatch<SetStateAction<EffortCurveForms | null>>;
+  selectedTractionMode: string | null;
+  isDefaultCurve: boolean;
+};
+
+const CurveSpreadsheet = ({
+  selectedCurve,
+  selectedCurveIndex,
+  effortCurves,
+  setEffortCurves,
+  selectedTractionMode,
+  isDefaultCurve,
+}: CurveSpreadsheetProps) => {
+  const { t } = useTranslation('rollingstock');
+
+  const spreadsheetCurve = useMemo(() => {
+    const { speeds, max_efforts } = selectedCurve.curve;
+    const filledMatrix: (
+      | {
+          value: string;
+        }
+      | undefined
+    )[][] =
+      speeds && max_efforts
+        ? max_efforts.map((effort, index) => [
+            {
+              value: speeds[index] !== undefined ? Math.round(speeds[index]! * 3.6).toString() : '',
+            },
+            { value: effort !== undefined ? Math.round(effort / 10).toString() : '' },
+          ])
+        : [];
+    const numberOfRows = filledMatrix.length < 8 ? 8 - filledMatrix.length : 1;
+    return filledMatrix.concat(createEmptyMatrix<CellBase<string>>(numberOfRows, 2));
+  }, [selectedCurve]);
+
+  const filterValidRows = (sheetValues: Matrix<{ value: string }>) =>
+    sheetValues.filter(
+      ([a, b]) =>
+        (a?.value && !emptyStringRegex.test(a.value)) ||
+        (b?.value && !emptyStringRegex.test(b.value))
+    );
+
+  const mapRowsToNumbers = (rows: Matrix<{ value: string }>) =>
+    rows.map((row) => row.map((cell) => onlyDigit((cell?.value || '').replaceAll(',', '.'))));
+
+  const reduceToCurveStructure = (mappedRows: Matrix<string>) =>
+    mappedRows.reduce<EffortCurveForm>(
+      (result, row) => {
+        result.speeds.push(row[0] !== '' ? Number(row[0]) / 3.6 : undefined);
+        result.max_efforts.push(row[1] !== '' ? Number(row[1]) * 10 : undefined);
+        return result;
+      },
+      { max_efforts: [], speeds: [] }
+    );
```

Comment:
this whole secton a lot of low level string/number manipulation. 
=> make functions with names that say what they do

Reply:
Done